### PR TITLE
Updateto o auth doc

### PR DESCRIPTION
**Pull request by wangxinlei**
**Things To Remember:** 

1. Be sure to merge into the right branch. The choices are:
* **`master`** if your PR runs on the current version of Tableau.
* **`dev-2021.2`**  
2. You should update at least one of the following:
* The changelog of the project you modified.
* The root `README.md` if the PR is a release of the SDK, Connector Packager, or TDVT.  
=====

### CODE REVIEW

File: docs/oauth.md [hunk 1]
```diff
@@ -211,6 +217,19 @@ tsm configuration set -k oauth.config.clients -v "[{\"oauth.config.id\":\"[your_
 Replace [your_dbclass] with the `dbclass` element registered in your oauthConfig.xml file. Substitute [your_client_id], [your_client_secret], [your_redirect_url] with the ones you registered in your provider's OAuth registration page.
 [your_redirect_url] needs to follow certain format, if your server address is https://Myserver/ then [your_reirect_uri] needs to be https://Myserver/auth/add_oauth_token.
 
+Tableau Online is managed by Tableau, in order for you connector to work on Tableau Online you will need to provide us(instruct us on how to create) the set of clientId/clientSecret/redirect_uri.
```
Comment (rosswbrown) [inline docs/oauth.md:220]:
```suggestion
Tableau Online is managed by Tableau, so in order for a connector to work on Tableau Online instructions to create a clientId, clientSecret, and redirect_uri will need to provided. 
```

File: docs/oauth.md [hunk 2]
```diff
@@ -211,6 +217,19 @@ tsm configuration set -k oauth.config.clients -v "[{\"oauth.config.id\":\"[your_
...
+## Site-Wide OAuth Clients
+Another way is through the site level OAuth client feature where the server admin for Tableau Server/ site admin for Tableau Online will be able to register the oauth client on a particular site, for example setting Azure AD oauth client on a site: https://help.tableau.com/current/server/en-us/config_oauth_azure_ad.htm.  
+The oauth clients will only be effective in the particular site, it did not require a restart and take precedence over the server wide oauth clients if any.
+
+| Server-Wide OAuth Clients  | Site-Wide OAuth Clients | 
+| ----  | ------- | 
+| tsm command running by server admin | User Interface and can be accessed by site admin(Tableau Online) or server admin(Tableau Server)|
+| Need restart server| No need to restart server|
+| Apply to whole server| Only apply to the particular site, will not affect other sites|
+| Low priority | High priority |
```
Comment (rosswbrown) [inline docs/oauth.md:231]: I don't understand this line.

File: docs/oauth.md [hunk 3]
```diff
@@ -96,6 +96,12 @@ Each OAuth config attribute is represented by an element in the XML, the element
 | accessTokenResponseMaps | Map<String, String> | Key value pair that maps an initial token request response attribute <value> to Tableau recognized attribute <key> | No | |
 | refreshTokenResponseMaps | Map<String, String> | Key value pair that maps an refresh token request response attribute <value> to Tableau recognized attribute <key> | Yes | If not defined will use accessTokenResponseMaps by default |
 
+
+> Note: For accessTokenResponseMaps, these following feilds are reserved for oauth flow and you need explicit map to these:  
+Required fields: `ACCESSTOKEN`(Used to Connector to your data), `REFRESHTOKEN`(Used to get a new ACCESSTOKEN), `username`(Used to identify the token by Tableau).  
+Optional fields: `access-token-issue-time` will default to the time when the token is sent back to Tableau if not present, `access-token-expires-in` will default to 3600s if not present.  
+Not listed fields: In tehory you can add any field as you like and maintain those, but they will not take effect in any oauth flow(initial auth, refresh token, get >user info, etc).
```
[the remark this answers is not in the document]
Comment (wangxinlei) [reply docs/oauth.md:103]: Until now no one need to do this. I think I am neutral to whether it's recommended, if they need to preserve something returned from oauth endpoint they can definitely do that via this way.

File: docs/oauth.md (hunk 2 above)
Comment (wangxinlei) [reply docs/oauth.md:231]: This is saying that the Site Level OAuth client will override the Server Level OAuth Client settings.

Comment (wangxinlei) [reply docs/oauth.md]: I've changed the wordings there, let me know if it's better.

File: docs/oauth.md [hunk 4]
```diff
@@ -96,6 +96,16 @@ Each OAuth config attribute is represented by an element in the XML, the element
 | accessTokenResponseMaps | Map<String, String> | Key value pair that maps an initial token request response attribute <value> to Tableau recognized attribute <key> | No | |
 | refreshTokenResponseMaps | Map<String, String> | Key value pair that maps an refresh token request response attribute <value> to Tableau recognized attribute <key> | Yes | If not defined will use accessTokenResponseMaps by default |
 
+Note: The keys in accessTokenResponseMaps and refreshTokenResponseMaps are tableau preserved field names and is defined in the followed table, the values are what your oauth provider returns in the raw response.
+| Name of the key | Optional/Required | Notes |
```
Comment (rosswbrown) [inline docs/oauth.md:100]: Have a Required column for accessTokenResponseMaps and one for refreshTokenResponseMaps. I think that will read more easily.

File: docs/oauth.md [hunk 5]
```diff
@@ -96,6 +96,16 @@ Each OAuth config attribute is represented by an element in the XML, the element
 | accessTokenResponseMaps | Map<String, String> | Key value pair that maps an initial token request response attribute <value> to Tableau recognized attribute <key> | No | |
 | refreshTokenResponseMaps | Map<String, String> | Key value pair that maps an refresh token request response attribute <value> to Tableau recognized attribute <key> | Yes | If not defined will use accessTokenResponseMaps by default |
 
+Note: The keys in accessTokenResponseMaps and refreshTokenResponseMaps are tableau preserved field names and is defined in the followed table, the values are what your oauth provider returns in the raw response.
```
Comment (rosswbrown) [inline docs/oauth.md:99]:
```suggestion
Note: The keys in accessTokenResponseMaps and refreshTokenResponseMaps are Tableau preferred field names and are defined in the followed table. The values are what your OAuth provider returns in the raw response.
```

File: docs/oauth.md [hunk 6]
```diff
@@ -211,6 +221,19 @@ tsm configuration set -k oauth.config.clients -v "[{\"oauth.config.id\":\"[your_
 Replace [your_dbclass] with the `dbclass` element registered in your oauthConfig.xml file. Substitute [your_client_id], [your_client_secret], [your_redirect_url] with the ones you registered in your provider's OAuth registration page.
 [your_redirect_url] needs to follow certain format, if your server address is https://Myserver/ then [your_reirect_uri] needs to be https://Myserver/auth/add_oauth_token.
 
+Tableau Online is managed by Tableau, so in order for a connector to work on Tableau Online instructions to create a clientId, clientSecret, and redirect_uri will need to provided.
+
+## Site-Wide OAuth Clients
+Another way is through the site level OAuth client feature where the server admin for Tableau Server/ site admin for Tableau Online will be able to register the oauth client on a particular site, for example setting Azure AD oauth client on a site: https://help.tableau.com/current/server/en-us/config_oauth_azure_ad.htm.
```
Comment (rosswbrown) [inline docs/oauth.md:227]:
```suggestion
Another way is through the site level OAuth client feature where the server admin for Tableau Server, or site admin for Tableau Online, will be able to register the OAuth client on a particular site. For example setting Azure AD OAuth client on a site: https://help.tableau.com/current/server/en-us/config_oauth_azure_ad.htm.  
```

**Review by rosswbrown — APPROVED** [review]

**Pull request merged**